### ISSUE TITLE
Add FilePicker in PlayStore variant.

### ISSUE THREAD

**Issue by MohitMaliFtechiz** [kiwix/kiwix-android#3628]
**Describe the bug**
While testing and refactoring the code for https://github.com/kiwix/kiwix-android/pull/3624, I was thinking of accessing the zim file with `file descriptor` without any permission(`MANAGE_EXTERNAL_STORAGE`). I can read the file without permission if I select the file via the existing `file picker` in the Play Store variant.

**Why we had failed to access the ZIM files in the past without special permission**
* We were trying to access the `File` which is the basic requirement of our application to function, and we can not access the files without special permission.
* We had some options at that time e.g. copying the content in a temp file from `inputStream` but it did not fit in our requirements, since it used twice the storage for the same ZIM file.
* With the `file descriptor` approach in https://github.com/kiwix/kiwix-android/issues/3265. In this PR we are trying to access the fileDescriptor from a file, which gives us the permission denied message for fileDescriptor. We did this with `old-libkiwix`
* In the past, we were using the `java-libkiwix` which had the issue opening the ZIM files via file descriptor that we saw while we were introducing the `ASSET_FILE_DELIVERY` for custom apps.

**Expected behavior**
We should add the `file picker` in the Play Store variant and if possible we should move our functionalities to rely on fileDescriptor instead of file. Since we have more access to a `file descriptor` instead of a file, Android limits direct access to files without special permission on Android 11 and above.

**What if we use file descriptor instead of file**
* It did not work for scanning the storage.
* It will work for opening the ZIM files with the file picker without any special permission.
* It needs refactoring to make the application properly work with the file descriptor.

**Additional notes to remember**
* We can show the ZIM file inside the library section if possible since at that time we have the URI of that ZIM file once opened with fileDescriptor.

https://github.com/kiwix/kiwix-android/assets/34593983/523d4503-3548-45aa-988b-ffc035ef6263

**Comment by Jaifroid** [comment]
@MohitMaliFtechiz This seems a very positive direction to me. Not being able to pick archives is the biggest disadvantage of the PlayStore variant of the Kiwix Android app (which, like it or not, is always going to be the main variant for the average user).

**Comment by kelson42** [comment]
Blocked by https://github.com/openzim/libzim/issues/852

**Comment by soloturn** [comment]
Following up on my earlier comment: the Play Store file-picker feature this PR (and #3628) set out to build has already shipped on `main`, just via a different approach that sidesteps the `libzim`/Xapian path limitation entirely.

`LocalLibraryViewModel.filePickerMenuButtonClick()` launches `ACTION_OPEN_DOCUMENT` (gated by `isPlayStoreBuildWithAndroid11OrAbove()`), and `ProcessSelectedZimFilesForPlayStore` (introduced in #4430) handles the picked URI by **copying/moving the selected ZIM into the app's own public directory**, then opening it as a normal `File` — rather than opening the archive directly via file descriptor/`/dev/fd/<n>`. That avoids the arm64/x86 `Permission denied` failure discussed above, since it never asks libzim to open an archive by fd path, and isn't blocked by openzim/libzim#852.

Given that, I don't think there's anything left to port forward here — recommend closing this PR (and #3628) as resolved by #4430, unless there's a reason fd-based direct opening (vs. copy-then-open) is still wanted for some other use case.

**Comment by soloturn** [comment]
Opened a draft fix for the underlying libzim issue: openzim/libzim#1119. It replaces the /dev/fd/<n> reopen-by-path with dup() of the fd already in hand. Draft pending maintainer review — see caveats in the PR description, in particular that I can't verify on real arm64/x86 Android hardware.

**Comment by soloturn** [comment]
Update on openzim/libzim#1119: reproduced the exact EACCES on a real arm64-v8a Android emulator with a minimal repro app -- reopening /dev/fd/<n> for a content-provider-backed fd fails with 'Permission denied', while dup() of the same fd succeeds. Details in the PR comment thread.

**Comment by Jaifroid** [comment]
Hmm, copying a 110GB file before opening it is likely to fail on all but the most well specified devices, so it's no substitute for the ability to pick a file directly and read it in situ, which even the PWA can do...

**Comment by soloturn** [comment]
> **AI-assisted comment.** Left by agent driven by @soloturn via [GDD](https://siliconsaga.github.io/yggdrasil/gdd/).

@Jaifroid correct. Copy/move (#4430) needs 2x storage. Fails on 110GB files. It's a fallback, not the real fix.

Root cause: openzim/libzim#852. Xapian needs its own fd. Old way: reopen `/dev/fd/<n>` by path. That reopen throws `EACCES` for some Android content-provider fds, on arm64-v8a/x86/x86_64. Same bug as #3636.

What `dup()` does: takes a fd you already have open, gives you a second fd number pointing at the same open file. No filesystem lookup, no path, no permission check — the OS just copies the existing open handle. Two independent fds, same underlying file, own seek position each. Can't fail with `EACCES` since nothing gets reopened.

Fix chain, all using `dup()` instead of reopen-by-path:

- **openzim/libzim#1119** (open, no review yet): `dup()` the fd already in hand. No path reopen, so no `EACCES`.
- **kiwix/java-libkiwix#151** (draft, needs #1119): same fd wired through JNI for large media items, not just Xapian.
- **kiwix/kiwix-android#5056** (draft, needs #151): `ZimFileReader` uses that fd. This is the actual read-in-situ path.
- **kiwix/kiwix-android#5057** (open, independent, mergeable now): separate pre-flight fd check was also reopening by path, silently rejecting good fds. Fixed same way.

Verified: reproduced the `EACCES` on a real arm64-v8a emulator, `dup()` fixes it there. x86/x86_64 not verified — no x86_64 emulator on this host. First three PRs chained, draft, pending #1119 merge + release. #5057 stands alone.
